lib/ValueAnonymizers/Email.php

```diff
@@ -25,7 +25,17 @@ public function anonymize(Value $value, array $row): AnonymizedValue
             return AnonymizedValue::fromOriginalValue($value);
         }
 
-        $anonymizedEscapedValue = $this->stringHash->hashKeepFormat($value->getUnEscapedValue());
+        $string = $value->getUnEscapedValue();
+
+        if (strlen($string) < 10) {
+            $hash = $this->stringHash->sha256($string);
+            $toAnonymize = substr($hash, 0, 10);
+            $toAnonymize .= '@'.substr($hash, 10, 5).'.'.substr($hash, 15, 3);
```

Comment:
as mainly the part before @ is the one bringing most of the entropy, I think we should make sure that part is bigger than a specific value ... 8 or 10.
This way we can keep identifying the domain length.

As another point, for email, I recommend that the domain part have a consistent hashing so we can identify the same domain on different anonymized emails, less important now.

Reply:
there is side problem to be handled if using domain hashes, as email fields can be as short as varchar 40